script/update-archive-version.ts

```diff
@@ -2,29 +2,48 @@
 import fs from 'fs';
 import path from 'path';
 import ProgressBar from 'progress';
-import { argv } from 'yargs';
+import argv from 'yargs';
 import { BookWithOSWebData } from '../src/app/content/types';
 import { makeUnifiedBookLoader } from '../src/app/content/utils';
+import { isDefined } from '../src/app/guards';
 import { ARCHIVE_URL, REACT_APP_ARCHIVE, REACT_APP_ARCHIVE_URL, REACT_APP_OS_WEB_API_URL } from '../src/config';
 import ArchiveUrlConfig from '../src/config.archive-url';
-import BOOKS from '../src/config.books';
+import BOOKS_CONFIG from '../src/config.books';
 import createArchiveLoader from '../src/gateways/createArchiveLoader';
 import createOSWebLoader from '../src/gateways/createOSWebLoader';
+import processBookVersionUpdate from './utils/update-content-versions';
 import updateRedirectsData from './utils/update-redirects-data';
 
 const configArchiveUrlPath = path.resolve(__dirname, '../src/config.archive-url.json');
 const { REACT_APP_ARCHIVE_URL_BASE } = ArchiveUrlConfig;
 
-const args = argv as any as {
-  newArchive: string
+const args = argv.string('pipelineVersion').argv as any as {
+  pipelineVersion: string,
+  contentVersion: string | string[],
 };
 
-async function updateArchiveVersion() {
-  if (args.newArchive === REACT_APP_ARCHIVE) {
-    console.log('Current and new archive url are the same. Skipping...');
+const getBooksToUpdate = (books: string[]) => books.map((book) => {
+  const bookId = book.split('@')[0];
+  const versionNumber = book.split('@')[1];
+  const { defaultVersion } = BOOKS_CONFIG[bookId] || {};
+  return defaultVersion === versionNumber ? undefined : {bookId, versionNumber};
+});
+
+async function updateArchiveAndContentVersions() {
+  const updatePipeline = args.pipelineVersion && args.pipelineVersion !== REACT_APP_ARCHIVE;
+  const booksReceived = args.contentVersion
+    ? (typeof args.contentVersion === 'string' ? [args.contentVersion] : args.contentVersion)
+    : [];
+  const booksToUpdate = booksReceived.length
+    ? getBooksToUpdate(booksReceived).filter(isDefined)
+    : [];
+
+  if (!updatePipeline && !booksToUpdate.length) {
+    console.log('Current and new archive url are the same. No books need version updates. Skipping...');
     return;
   }
 
+  const newArchiveUrl = `${REACT_APP_ARCHIVE_URL_BASE}${args.pipelineVersion}`;
```

Comment:
oh i see because for the non-pipeline update you construct a different one [here](https://github.com/openstax/rex-web/pull/1418/files#diff-36084040e5187727e1bf828bd23b4a032f998045a260bbb8f630306b0a38c9d5R16)... hmm... you could probably just move that here and pass it in, its almost duplicated code